lib/active_resource/detailed_log_subscriber.rb

```diff
@@ -21,17 +23,25 @@ def log_request_response_details(event)
       info "Response:\n#{event.payload[:response].body}"
     end
 
-    def warn_on_deprecated_header(event)
+    def warn_on_deprecated_header_or_version_eol_header(event)
       payload = event.payload
 
       payload[:response].each do |header_name, header_value|
         case header_name.downcase
-        when 'x-shopify-api-deprecated-reason'
+        when VERSION_DEPRECATION_HEADER
           warning_message = <<-MSG
           [DEPRECATED] ShopifyAPI made a call to #{payload[:method].upcase} #{payload[:path]}, and this call made
           use of a deprecated endpoint, behaviour, or parameter. See #{header_value} for more details.
           MSG
 
+          warn warning_message
+
+        when VERSION_EOL_WARNING_HEADER
+          warning_message = <<-MSG
+          [Version EOL Warning] ShopifyAPI made a call to #{payload[:method].upcase} #{payload[:path]}, and this call made
```

Comment:
We shouldn't use EOL without defining it and I'm not even sure we publicly use that term.

@chrisbutcher isn't it just "unsupported"? We just need to make sure this lines up with our public versioning terminology.

Reply:
Yes, the term we should use publicly is `unsupported`: https://shopify.dev/concepts/about-apis/versioning